### PR TITLE
feat: add v9 of fox farm

### PR DESCRIPTION
**Pull request by 0xean**
## Description

Adds v9 of fox farming.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

should be considered high risk even tho a tiny PR just in case I put the wrong addresses in or something of that nature. 

> What protocols, transaction types or contract interactions might be affected by this PR?

ETH fox farming 

## Testing

Not tested besides to confirm it shows up on screen. Ops might be able to test before it goes live with some small LP balances just to confirm staking and unstaking works and flows into the new contracts correctly 

https://etherscan.io/address/0xec84524385ffd1ea423ec4682caa5ab56071fbc7

### Operations

Please give this a once over with staking and unstaking to confirm funds flow into and out of 

https://etherscan.io/address/0xec84524385ffd1ea423ec4682caa5ab56071fbc7

## Screenshots (if applicable)
Screenshot 2024-05-08 at 2 51 19 PM

### CODE REVIEW

**Comment by 0xean** [comment]
P.S I updated the arkeo code, just because it was there, but creating follow up ticket to remove https://app.shapeshift.com/#/arkeo entirely as the snapshot is now down

**Review by gomesalexandre — APPROVED** [review]
Tested locally, confirmed this does what it says on the box!

image

**Review by woodenfurniture — CHANGES_REQUESTED** [review]
Unsure whether this is intentional but out of caution, requestiong changes so we can verify. Clicking the main "Start earning" appears to be linked to v8, as it shows the v8 contract address instead of the v9 contract address.

image

Uploading Screen Recording 2024-05-10 at 9.33.43 AM.mov…

**Comment by woodenfurniture** [comment]
V9 was not appearing in the carousel due to a filter enforcing >1% APY and >$50k TVL, which obviously the new contract does not satisfy. I've pushed a commit to hardcode the v9 contract to appear at the front of the carousel which we cna revert once V9 has adequate TVL.

**Review by woodenfurniture — APPROVED** [review]

File: src/pages/Defi/components/EligibleCarousel.tsx [hunk 1]
```diff
@@ -37,10 +38,28 @@ export const EligibleCarousel: React.FC<EligibleCarouselProps> = props => {
...
+    if (!foxFarmingV9) {
+      console.error('Fox Farming V9 opportunity not found in eligible opportunities')
+      return filteredEligibleOpportunities
+    }
+
+    // TEMP: Hardcode the Fox Farming V9 opportunity to be the first card until enough TVL is in the pool
+    const filteredEligibleOpportunitiesWithFoxFarmingV9 = [
+      foxFarmingV9,
+      ...filteredEligibleOpportunities,
+    ].slice(0, 5)
```
Comment (0xApotheosis) [inline src/pages/Defi/components/EligibleCarousel.tsx:60]: Add a uniqueBy for when `foxFarmingV9` appears in the regular list.

File: src/pages/Defi/components/EligibleCarousel.tsx [hunk 2]
```diff
@@ -37,10 +39,27 @@ export const EligibleCarousel: React.FC<EligibleCarouselProps> = props => {
...
+        eligibleOpportunity.contractAddress === ETH_FOX_STAKING_CONTRACT_ADDRESS_V9,
+    )
+
+    if (!foxFarmingV9) {
+      return filteredEligibleOpportunities
+    }
+
+    // TEMP: Hardcode the Fox Farming V9 opportunity to be the first card until enough TVL is in the pool
+    const filteredEligibleOpportunitiesWithFoxFarmingV9 = uniqBy(
+      [filteredEligibleOpportunities[0], foxFarmingV9, ...filteredEligibleOpportunities.slice(1)],
```
Comment (0xApotheosis) [inline src/pages/Defi/components/EligibleCarousel.tsx:58]: This is gross, but as discussed, it'll do the trick.

**Review by 0xApotheosis — APPROVED** [review]
The spread is pretty gnarly, but I get why we are doing it.

V9 shows as expected, and initial load with no cache issue is mitigated with the re-positioning of the V9 contract as the second element of the opportunities list.

**Pull request merged**